feat:(Add submenu option)

danlaush opened the pull request:
~Note: This is a work in progress—PR created for discussion.~ PR ready for review now.

The goal of this work is to allow public-navigation integrations to include a 2nd-level menu, such as appears on the [IBAN checker](https://transferwise.com/gb/iban/) and [blog](https://transferwise.com/gb/blog/). 

Example:
![Screenshot 2019-09-26 at 12 43 29](https://user-images.githubusercontent.com/1204500/65686416-41f99700-e05c-11e9-9a31-fe6b5b2bd022.png)

There's a couple things I think would be good to discuss/get agreement on before I go much further on this.

- Approach, 2 ideas:
    - Allow the consuming application to pass in the items to be shown in the second-level nav (see Localisation) 
    - Add the menu structure for each project using the public-navigation and allow the consuming application to pass in a key that corresponds to the correct submenu items.
- Localisation
    - @brettanomyces suggested that it might be easier/more maintainable for integrators to, instead of adding translation keys to public-navigation for each project's use case, pass in links which have already been localised in the consuming application. This allows public-navigation to be agnostic and doesn't require an update when the application requires a change in the menu, which would (presumably) be in a different team.
- Styles
    - I've made a stab at using the Bootstrap submenu classes to add the second-level nav, see below. This is how the existing IBAN checker et al. implement this. There's a few little adjustments that I think will need to be made, as there's some things in there like `class="navbar--inverse navbar-inverse"` (the former from the react component, the latter from bootstrap) and some other little things caused by overlapping CSS.
    - It might be easiest to work through these on a case-by-case basis, but I'm still figuring out so it would be great to chat through it with someone who has more history on it.

Current effort (26 Sep):
![Screenshot 2019-09-25 at 15 51 21](https://user-images.githubusercontent.com/1204500/65686381-2c846d00-e05c-11e9-979a-38750988f84e.png)

danlaush commented on docs/Docs.less:
```diff
@@ -8,7 +8,7 @@
   right: 0;
 
-  height: 56px;
+  height: 64px;
```
Per designs, when the nav has a submenu it needs to be taller. Can be handled in JS logic.

mikknarusk commented:
@danlaush I think this approach makes sense. Let's allow applications to pass in a submenu array, with strings already translated. I assume we should still pass in unlocalised urls (eg `/:locale/borderless` as it doesn't require any configuration on the public-nav side and works out of the box. Also keeps the logic of mapping locales to urls in a single place.

Nice job, looks easy enough, although mobile could require more fine tuning :)

danlaush commented:
@mikknarusk This is now ready for review! The git history got a bit cooked when I rebased, unfortunately. 🤔 Hate keeping branches open for so long.

![Screenshot 2019-10-31 at 15 08 35](https://user-images.githubusercontent.com/1204500/67959262-59deb080-fbf0-11e9-9434-ce6b47bd10d6.png)

fergusjordantw commented on src/PublicNavigation/Navigation/Navigation.js:
```diff
@@ -37,9 +38,27 @@ class Navigation extends Component {
+          className,
+        ]
+          .filter(i => i) // remove falseys
+          .join(' ')}
```
Can we just use `classNames` here instead as it's already a project dependency?

fergusjordantw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,15 @@
+# v3.4.5
+## Added support for subnav
```
Should this be a minor version bump?

danlaush commented:
@mikknarusk @fergusjordantw I'm closing this PR in favour of https://github.com/transferwise/public-navigation/pull/121 where I cleaned up the commit history. Fergus, I've addressed all your comments here in that branch.
